### PR TITLE
Fail when adding nodes to a CLB results in a node limit error

### LINKED ISSUE #1214: AddNodesToCLB should fail on 25 nodes limit
It should return `StepResult.FAILURE` when 25 nodes limit error occurs since group cannot scale anymore after that. Can be done after #1189.

### LINKED ISSUE #1244: CLBChangeNode should handle over limit errors
Unfortunately, it appears that 413 (normally for requested entity too large) is used for several failure modes, specifically "too many nodes on the CLB" and "too many requests".

```
[11:22:19] <lvh> which one of those errors do I get back if the LB has too many nodes?
[11:22:38] <lvh> the obvious one is  overLimit (413) but IIRC that's also used if you're issuing too many requests
[11:27:33] <SCO_bmetzler> lvh:  There is no direct way to figure that out.  The only way to figure this out is to pull the Rate Limits programmatically and test against those.  http://docs.rackspace.com/loadbalancers/api/v1.0/clb-devguide/content/Rate_Limits-d1e821.html
```

#1214 exists to handle not issuing the requests that cause this in the first place, which leads me to believe that the correct resolution for *this* ticket is to just retry.

### LINKED ISSUE #1260: Remove `get_fake_service_request_performer` from otter/test/utils.py
Hopefully, service request performing will just be done in the client (see #1189), and if so, this may no longer be a necessary general utility.

After which it can be removed.

### PR DESCRIPTION
**Pull request by cyli**
If the are more 25 nodes on a CLB and convergence tries to add more, then the step should fail and convergence should put the group into ERROR.

Since we're touching the add nodes code anyway, I moved it to cloud_client as per #1189.

This makes it so that all 4XX errors, unless we know we can retry, fail.  Other errors are retried.

Fixes #1214.  
Fixes #1244.
Fixes #1260.

### CODE REVIEW

**Comment by cyli** [comment]
jenkins please run integration tests

**Comment by cyli** [comment]
jenkins please run integration tests

**Comment by cyli** [comment]
Actually, I'm going to try to get #1524 merged first, since I'm going to need those for another PR.

**Comment by cyli** [comment]
Ok, this is probably ready for review again.  #1513 still passes with master merged in and changes.

**Comment by cyli** [comment]
jenkins please run integration tests

File: otter/cloud_client.py [hunk 1]
```diff
@@ -416,6 +439,52 @@ def try_parsing(api_error_exc_info):
...
+        }
+
+        (weight and type are optional)
+
+    :return: :class:`ServiceRequest` effect
+
+    :raises: :class:`CLBPendingUpdateError`, :class:`CLBDeletedError`,
+        :class:`NoSuchCLBError`, :class:`CLBDuplicateNodesError`,
+        :class:`APIError`
+    """
```
Comment (manishtomar) [inline otter/cloud_client.py]: Great docs :)

File: otter/convergence/steps.py [hunk 2]
```diff
@@ -350,18 +341,22 @@ def report_api_failure(result):
...
-                    ErrorReason.Exception(result)]
-            if error.code == 422:
-                message = try_json_with_keys(error.body, ("message",))
-                if message and _CLB_DELETED_PATTERN.search(message):
-                    return StepResult.FAILURE, [
-                        ErrorReason.Exception(result)]
-            return StepResult.RETRY, [ErrorReason.Exception(result)]
+            status = StepResult.FAILURE
+
+            known_retryable = err_type in (CLBDuplicateNodesError,
```
Comment (radix) [inline otter/convergence/steps.py]: I think maybe with a few small abstractions we could simplify this function a lot, but that's just a side thought. LGTM :)

**Comment by radix** [comment]
Great changes to the unit tests, too. Thanks a lot!

**Pull request merged**